pkg/network/events/monitor_windows.go

```diff
@@ -0,0 +1,79 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// Package events handles process events
+package events
+
+import (
+	"os"
+	"path/filepath"
+	"time"
+
+	"go4.org/intern"
+
+	"github.com/DataDog/datadog-agent/pkg/security/secl/model"
+	"github.com/DataDog/datadog-agent/pkg/util/winutil/iisconfig"
+)
+
+func getProcessStartTime(ev *model.Event) time.Time {
+	if ev.GetEventType() == model.ExecEventType {
+		return ev.GetProcessExecTime()
+	}
+	// windows does not have a fork event.
+	return time.Time{}
+}
+
+func makeTagsSlice(apmtags iisconfig.APMTags) []*intern.Value {
+	tags := make([]*intern.Value, 0, 3)
+	if len(apmtags.DDService) > 0 {
+		tags = append(tags, intern.GetByString("service"+":"+apmtags.DDService))
+	}
+	if len(apmtags.DDEnv) > 0 {
+		tags = append(tags, intern.GetByString("env"+":"+apmtags.DDEnv))
+	}
+	if len(apmtags.DDVersion) > 0 {
+		tags = append(tags, intern.GetByString("version"+":"+apmtags.DDVersion))
+	}
+	if len(tags) == 0 {
+		return nil
+	}
+	return tags
+}
+func getAPMTags(filename string) []*intern.Value {
+
+	dir := filepath.Dir(filename)
+	if dir == "" {
+		return nil
+	}
+
+	// see if there's an app.config in the directory
+	appConfig := filepath.Join(dir, "app.config")
+	ddJSON := filepath.Join(dir, "datadog.json")
+	if _, err := os.Stat(appConfig); err == nil {
+
+		appcfg, err := iisconfig.ReadDotNetConfig(appConfig)
+		if err == nil {
+			found := makeTagsSlice(appcfg)
+			if len(found) > 0 {
+				return found
+			}
+		}
+	}
+	// see if there's a datadog.json
+	if _, err := os.Stat(ddJSON); err == nil {
+
+		appcfg, err := iisconfig.ReadDatadogJSON(ddJSON)
+		if err == nil {
+			if err == nil {
+				found := makeTagsSlice(appcfg)
+				if len(found) > 0 {
+					return found
+				}
+			}
+		}
+	}
```

Comment:
Couple questions for this block:

1. Is it intentional that the app config content is given precedence over the datadog.json content? Is that documented or did you just happen to choose to parse the app config first? If the order matters, we should confirm that this is the right hierarchy (maybe we already did).
2. For these files, are all environment variables expected to be in each file or could they be split? Right now, if they were split between the files, we'd ignore anything in datadog.json if any (but not all) were defined in the app config
3. Related to (2.) - if there is an app config and at least one environment variable in it, we don't look for datadog.json. In my opinion, we should always check for and parse a datadog.json if it is there, unless we've established and documented to customer that they can only use one or the other AND all environments must be in which ever file they pick.

If we haven't made that clear, we should. Right now, I think for this we should figure out which file has precedence and parse both, with a tie going to whichever we decide has precedence and document it. I suspect since this is new, it isn't documented, so let's decide now and stick with it.

Reply:
1. Yes, it's intentional.  Yes it's documented (in the internal doc)
2. I'm not sure what the expected behavior would be in that case, but it seems pathological.  We can check with APM to see how to handle
3. there are no "environment variables" in app config.  app config is allowed to have one of the settings which is essentially a "redirect" to a `datadog.json` file.

Reply:
1. All good
2. All good
3. What do you mean by there aren't any environment variables? in the `ReadDotNetConfig` function (which I guess is actually reading the `AppConfig`) we look for `DD_` entries, aren't those the "environment variables" that customers define that would normally be actually environment variables on the process?

Reply:
those are the key names that happen to be the same as the environment variables.

Reply:
Thanks for explaining that a process's environment variables don't live in JSON files.

What I meant were the keys in the JSON files that map to environment variables, but I guess what was confusing.

So, question / comment (3.) rewritten to be clearer to you:

Related to (2.) - if there is an app config and at least one ~~environment variable~~ key, we don't look for datadog.json. In my opinion, we should always check for and parse a datadog.json if it is there, unless we've established and documented to customer that they can only use one or the other AND all ~~environments~~ must be in which ever file they pick.

Hopefully that makes more sense.

Reply:
I have checked with the APM team and you're right.  I'll update the doc to cover that case, and the code.